src/core/mempool.rs

```diff
@@ -523,118 +618,105 @@ impl MemPoolDB {
         F: FnMut(Vec<MemPoolTxInfo>) -> Result<(), E>,
         E: From<db_error> + From<ChainstateError>,
     {
-        let (mut tip_consensus_hash, mut tip_block_hash, mut tip_height) = (
+        let (
+            mut curr_tip_consensus_hash,
+            max_tip_consensus_hash,
+            mut curr_tip_block_hash,
+            max_tip_block_hash,
+            max_tip_height,
+            mut curr_tip_height,
+        ) = (
             tip_consensus_hash.clone(),
+            tip_consensus_hash.clone(),
+            tip_block_hash.clone(),
             tip_block_hash.clone(),
             tip_height,
+            None,
```

Comment:
Something that could make this process a little more efficient (only a little, because the most costly part of the walk is the MARF lookups done in `get_index_tip_ancestor_conn`, which I think are avoided if a given block height doesn't have any transactions in the mempool) is to initialize `curr_tip_height` to `Some(tip_height - MEMPOOL_MAX_TRANSACTION_AGE)` (you'd want to check for underflow on that).